<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="theme-color" content="#222"><meta name="generator" content="Hexo 7.2.0">

  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css" integrity="sha256-wiz7ZSCn/btzhjKDQBms9Hx4sSeUYsDrTLg7roPstac=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.1.1/animate.min.css" integrity="sha256-PR7ttpcvz8qrF57fur/yAx1qXMFJeJFiA6pSzWi0OIE=" crossorigin="anonymous">

<script class="next-config" data-name="main" type="application/json">{"hostname":"louis-kento.github.io","root":"/","images":"/images","scheme":"Muse","darkmode":false,"version":"8.19.2","exturl":false,"sidebar":{"position":"left","width_expanded":320,"width_dual_column":240,"display":"post","padding":18,"offset":12},"copycode":{"enable":false,"style":null},"fold":{"enable":false,"height":500},"bookmark":{"enable":false,"color":"#222","save":"auto"},"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"stickytabs":false,"motion":{"enable":true,"async":false,"transition":{"menu_item":"fadeInDown","post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}}</script><script src="/js/config.js"></script>

    <meta name="description" content="#####miss0.15cmatrix -B -C blue -u 9 sed -e ‘s&#x2F;2&#x2F;NC_019459.2&#x2F;g’ -e ‘s&#x2F;1&#x2F;NC_019458.2&#x2F;g’ -e ‘s&#x2F;3&#x2F;NC_019460.2&#x2F;g’ -e ‘s&#x2F;4&#x2F;NC_019461.2&#x2F;g’ -e ‘s&amp;#">
<meta property="og:type" content="article">
<meta property="og:title" content="Hexo">
<meta property="og:url" content="https://louis-kento.github.io/2024/04/28/fst%E5%9F%BA%E7%A1%80/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:description" content="#####miss0.15cmatrix -B -C blue -u 9 sed -e ‘s&#x2F;2&#x2F;NC_019459.2&#x2F;g’ -e ‘s&#x2F;1&#x2F;NC_019458.2&#x2F;g’ -e ‘s&#x2F;3&#x2F;NC_019460.2&#x2F;g’ -e ‘s&#x2F;4&#x2F;NC_019461.2&#x2F;g’ -e ‘s&amp;#">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2024-04-28T12:24:54.995Z">
<meta property="article:modified_time" content="2023-08-25T12:31:14.000Z">
<meta property="article:author" content="John Doe">
<meta name="twitter:card" content="summary">


<link rel="canonical" href="https://louis-kento.github.io/2024/04/28/fst%E5%9F%BA%E7%A1%80/">



<script class="next-config" data-name="page" type="application/json">{"sidebar":"","isHome":false,"isPost":true,"lang":"en","comments":true,"permalink":"https://louis-kento.github.io/2024/04/28/fst%E5%9F%BA%E7%A1%80/","path":"2024/04/28/fst基础/","title":""}</script>

<script class="next-config" data-name="calendar" type="application/json">""</script>
<title> | Hexo</title>
  








  <noscript>
    <link rel="stylesheet" href="/css/noscript.css">
  </noscript>
</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <div class="column">
      <header class="header" itemscope itemtype="http://schema.org/WPHeader"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <p class="site-title">Hexo</p>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger" aria-label="Search" role="button">
    </div>
  </div>
</div>







</header>
        
  
  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%A0%B9%E6%8D%AELD%E6%9D%A5%E8%BF%87%E6%BB%A4%E8%BF%9E%E9%94%81%E7%9A%84SNP"><span class="nav-number">1.</span> <span class="nav-text">根据LD来过滤连锁的SNP</span></a></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">John Doe</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
        <a href="/archives/">
          <span class="site-state-item-count">3</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
  </nav>
</div>

        </div>
      </div>
    </div>

    
  </aside>


    </div>

    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="https://louis-kento.github.io/2024/04/28/fst%E5%9F%BA%E7%A1%80/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="John Doe">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Hexo">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="post" itemscope itemtype="http://schema.org/CreativeWork">
      <meta itemprop="name" content=" | Hexo">
      <meta itemprop="description" content="">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2024-04-28 20:24:54" itemprop="dateCreated datePublished" datetime="2024-04-28T20:24:54+08:00">2024-04-28</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar-check"></i>
      </span>
      <span class="post-meta-item-text">Edited on</span>
      <time title="Modified: 2023-08-25 20:31:14" itemprop="dateModified" datetime="2023-08-25T20:31:14+08:00">2023-08-25</time>
    </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody"><p>#####miss0.15<br>cmatrix -B -C blue -u 9</p>
<p>sed -e ‘s&#x2F;2&#x2F;NC_019459.2&#x2F;g’ -e ‘s&#x2F;1&#x2F;NC_019458.2&#x2F;g’ -e ‘s&#x2F;3&#x2F;NC_019460.2&#x2F;g’ -e ‘s&#x2F;4&#x2F;NC_019461.2&#x2F;g’ -e ‘s&#x2F;5&#x2F;NC_019462.2&#x2F;g’ -e ‘s&#x2F;6&#x2F;NC_019463.2&#x2F;g’ -e ‘s&#x2F;7&#x2F;NC_019464.2&#x2F;g’ -e ‘s&#x2F;8&#x2F;NC_019465.2&#x2F;g’ -e ‘s&#x2F;9&#x2F;NC_019466.2&#x2F;g’ -e ‘s&#x2F;10&#x2F;NC_019467.2&#x2F;g’ -e ‘s&#x2F;11&#x2F;NC_019468.2&#x2F;g’ -e ‘s&#x2F;12&#x2F;NC_019469.2&#x2F;g’ -e ‘s&#x2F;13&#x2F;NC_019470.2&#x2F;g’ -e ‘s&#x2F;14&#x2F;NC_019471.2&#x2F;g’ -e ‘s&#x2F;15&#x2F;NC_019472.2&#x2F;g’ -e ‘s&#x2F;16&#x2F;NC_019473.2&#x2F;g’ -e ‘s&#x2F;17&#x2F;NC_019474.2&#x2F;g’ -e ‘s&#x2F;18&#x2F;NC_019475.2&#x2F;g’ -e ‘s&#x2F;19&#x2F;NC_019476.2&#x2F;g’ -e ‘s&#x2F;20&#x2F;NC_019477.2&#x2F;g’  -e ‘s&#x2F;21&#x2F;NC_019478.2&#x2F;g’ -e ‘s&#x2F;22&#x2F;NC_019479.2&#x2F;g’ -e ‘s&#x2F;23&#x2F;NC_019480.2&#x2F;g’ -e ‘s&#x2F;24&#x2F;NC_019481.2&#x2F;g’ -e ‘s&#x2F;25&#x2F;NC_019482.2&#x2F;g’ -e ‘s&#x2F;26&#x2F;NC_019483.2&#x2F;g’ -e ‘s&#x2F;27&#x2F;NC_019484.2&#x2F;g’ -e ‘s&#x2F;mt&#x2F;NC_001941.1&#x2F;g’  re8.vcf.recode.vcf &gt; husitef.vcf</p>
<p>#单点                                </p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..8&#125;;<span class="keyword">do</span> vcftools --vcf /mnt/d/lx/soft/VCF_add_id/cid.vcf --weir-fst-pop /mnt/d/lx/data/grouph/group19.txt --fst-window-size 100000 --fst-window-step 50000 --weir-fst-pop /mnt/d/lx/data/grouph/group<span class="variable">$i</span>.txt  --out hu_<span class="variable">$i</span> ;<span class="keyword">done</span> </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;9..16&#125;;<span class="keyword">do</span> vcftools --vcf /mnt/d/lx/soft/VCF_add_id/cid.vcf --weir-fst-pop /mnt/d/lx/data/grouph/group19.txt --fst-window-size 100000 --fst-window-step 50000 --weir-fst-pop /mnt/d/lx/data/grouph/group<span class="variable">$i</span>.txt  --out hu_<span class="variable">$i</span> ;<span class="keyword">done</span> </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;17..24&#125;;<span class="keyword">do</span> vcftools --vcf /mnt/d/lx/soft/VCF_add_id/cid.vcf --weir-fst-pop /mnt/d/lx/data/grouph/group19.txt --fst-window-size 100000 --fst-window-step 50000 --weir-fst-pop /mnt/d/lx/data/grouph/group<span class="variable">$i</span>.txt  --out hu_<span class="variable">$i</span> ;<span class="keyword">done</span> </span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;25..31&#125;;<span class="keyword">do</span> vcftools --vcf /mnt/d/lx/soft/VCF_add_id/cid.vcf --weir-fst-pop /mnt/d/lx/data/grouph/group19.txt --fst-window-size 100000 --fst-window-step 50000 --weir-fst-pop /mnt/d/lx/data/grouph/group<span class="variable">$i</span>.txt  --out hu_<span class="variable">$i</span> ;<span class="keyword">done</span> </span><br><span class="line">vcftools --vcf /mnt/d/lx/soft/VCF_add_id/cid.vcf --weir-fst-pop /mnt/d/lx/data/grouph/group19.txt --weir-fst-pop /mnt/d/lx/data/grouph/group23.txt  --out hu_23</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">vcftools --vcf /home/l/data/vcf/anno/150kb/6_28-30/28-30.vcf --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/hu11.txt --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/feizhou.txt  --fst-window-size 5000 --out hu_feizhou.fst </span><br><span class="line">vcftools --vcf /home/l/data/vcf/anno/150kb/6_28-30/28-30.vcf --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/hu11.txt --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/yeyang.txt --fst-window-size 5000 --out hu_yeyang.fst </span><br><span class="line">vcftools --vcf /home/l/data/vcf/anno/150kb/6_28-30/28-30.vcf --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/hu11.txt --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/305.txt --fst-window-size 5000 --out hu_all.fst </span><br><span class="line">vcftools --vcf /home/l/data/vcf/anno/150kb/6_28-30/28-30.vcf --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/hu11.txt --weir-fst-pop /home/l/data/vcf/anno/150kb/6_28-30/yinjin.txt --fst-window-size 5000 --out hu_yinjin.fst </span><br></pre></td></tr></table></figure>

<p>cat *fst &gt; m.txt<br>awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m1.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; mmmm.txt &amp;&amp; head -n 9025 mmmm.txt &gt; top1.txt</p>
<p>awk ‘{if($6~&#x2F;e&#x2F;)$6&#x3D;0}1’ w100hu_1.fst.windowed.weir.fst &gt; hu_am.txt</p>
<p>awk ‘{if($8<del>&#x2F;e&#x2F;)$8&#x3D;0}1’ 6.txt &gt; 66.txt awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ 2.txt &gt; 3.txt</p>
<p>sort -nr -k3 one.yj.fst &gt; cs_hu_yj.fst<br>sort -nr -k3 one.df.fst &gt; cs_hu_df.fst</p>
<p>sort -nr -k6 hu_am.txt &gt; sort.txt</p>
<p>#窗口</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..31&#125;; <span class="keyword">do</span> sed -e <span class="string">&#x27;s/NC_019459.2/2/g&#x27;</span> -e <span class="string">&#x27;s/NC_019458.2/1/g&#x27;</span> -e <span class="string">&#x27;s/NC_019460.2/3/g&#x27;</span> -e <span class="string">&#x27;s/NC_019461.2/4/g&#x27;</span> -e <span class="string">&#x27;s/NC_019462.2/5/g&#x27;</span> -e <span class="string">&#x27;s/NC_019463.2/6/g&#x27;</span> -e <span class="string">&#x27;s/NC_019464.2/7/g&#x27;</span> -e <span class="string">&#x27;s/NC_019465.2/8/g&#x27;</span> -e <span class="string">&#x27;s/NC_019466.2/9/g&#x27;</span> -e <span class="string">&#x27;s/NC_019467.2/10/g&#x27;</span> -e <span class="string">&#x27;s/NC_019468.2/11/g&#x27;</span> -e <span class="string">&#x27;s/NC_019469.2/12/g&#x27;</span> -e <span class="string">&#x27;s/NC_019470.2/13/g&#x27;</span> -e <span class="string">&#x27;s/NC_019471.2/14/g&#x27;</span> -e <span class="string">&#x27;s/NC_019472.2/15/g&#x27;</span> -e <span class="string">&#x27;s/NC_019473.2/16/g&#x27;</span> -e <span class="string">&#x27;s/NC_019474.2/17/g&#x27;</span> -e <span class="string">&#x27;s/NC_019475.2/18/g&#x27;</span> -e <span class="string">&#x27;s/NC_019476.2/19/g&#x27;</span> -e <span class="string">&#x27;s/NC_019477.2/20/g&#x27;</span>  -e <span class="string">&#x27;s/NC_019478.2/21/g&#x27;</span> -e <span class="string">&#x27;s/NC_019479.2/22/g&#x27;</span> -e <span class="string">&#x27;s/NC_019480.2/23/g&#x27;</span> -e <span class="string">&#x27;s/NC_019481.2/24/g&#x27;</span> -e <span class="string">&#x27;s/NC_019482.2/25/g&#x27;</span> -e <span class="string">&#x27;s/NC_019483.2/26/g&#x27;</span> -e <span class="string">&#x27;s/NC_019484.2/x/g&#x27;</span> -e <span class="string">&#x27;s/NC_001941.1/mt/g&#x27;</span> w50hu_<span class="variable">$i</span>.fst.windowed.weir.fst &gt; change<span class="variable">$i</span>.fst ; <span class="keyword">done</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..31&#125;; <span class="keyword">do</span> awk <span class="string">&#x27;&#123;if($3~/e/)$3=0&#125;1&#x27;</span> w50_25hu_<span class="variable">$i</span>.fst.windowed.weir.fst &gt; one<span class="variable">$i</span>.fst  ; <span class="keyword">done</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..31&#125;; <span class="keyword">do</span> <span class="built_in">sort</span> -nr -k6 one<span class="variable">$i</span>.fst &gt; cs<span class="variable">$i</span>.fst ; <span class="keyword">done</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..31&#125;; <span class="keyword">do</span> <span class="built_in">head</span> -n 1006 cs<span class="variable">$i</span>.fst &gt; top<span class="variable">$i</span>.fst ; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>



<p>for i in {1..31}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ &#x2F;home&#x2F;l&#x2F;data&#x2F;fst_window&#x2F;0.2&#x2F;50-25&#x2F;top$i.fst &gt; top$i.txt ; done<br>for i in {1..31}; do awk ‘{if($6~&#x2F;e&#x2F;)$6&#x3D;0}1’ hu_&amp;i.weir.fst &gt; one$i.fst  ; done<br>for i in {1..31}; do sort -nr -k3 one$i.fst &gt; cs$i.fst ; done<br>sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ sort.tajimaD &gt; use.tajimaD</p>
<p>for i in {1..31}; do awk ‘{if($3~&#x2F;e&#x2F;)$3&#x3D;0}1’ hu_$i.weir.fst &gt; one$i.fst &amp;&amp; sort -nr -k3 one$i.fst &gt; cs$i.fst &amp;&amp; head -n 2980 cs$i.fst &gt; top$i.txt ; done</p>
<p>for i in {1..31}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ w100hu_$i.fst.windowed.weir.fst &gt; change$i.fst ; done<br>for i in {1..31}; do awk ‘{if($6~&#x2F;e&#x2F;)$6&#x3D;0}1’ change$i.fst &gt; one$i.fst  ; done<br>for i in {1..31}; do sort -nr -k6 one$i.txt &gt; cs$i.txt ; done<br>对文件第2列做减1处理，方便后面与TajimaD匹配</p>
<p>for i in {1..31}; do head -n 1707 cs$i.fst &gt; 150top$i.txt ; done</p>
<p>awk ‘{$2 &#x3D; $2-1;print$0}’<br>25351<br>for i in {1..31}; do head -n 2535 use$i.fst &gt; 100top$i.fst ; done<br>head -n 1213 use.tajimaD &gt; top0.05.tajimaD</p>
<p>tail -n 1213 use.tajimaD &gt; bot0.05.tajimaD</p>
<p>for i in {1..31}; do cat &#x2F;home&#x2F;l&#x2F;data&#x2F;fst_window&#x2F;0.2&#x2F;100kb&#x2F;bot0.05.tajimaD &#x2F;home&#x2F;l&#x2F;data&#x2F;fst_window&#x2F;0.2&#x2F;100kb&#x2F;100top$i.fst &gt; DA$i.bot &amp;&amp; awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;1) print }’ pass&#x3D;1 DA$i.bot pass&#x3D;2 DA$i.bot &gt; DA$i.bot.site ; done</p>
<p>for i in {1..5}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>for i in {6..10}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>for i in {11..15}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>for i in {16..20}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>for i in {21..25}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>for i in {26..31}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;miss0.8.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;w100hu_$i.fst ;done<br>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –window-pi 500000 –window-pi-step 25000 –out pi_50kb_25kb<br>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –TajimaD 75000  –out tajimad_75<br>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –hardy –max-missing 1.0 –out all_snp_hardy<br>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –weir-fst-pop hu11.txt –weir-fst-pop yinjin.txt  –fst-window-size 100000  –out w100hu_yinjin.fst</p>
<p>java -jar snpEff.jar databases | grep “ovis”<br>vcftools  –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf –plink –out output<br>plink –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf –recode –out file<br>plink –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf –recode –out pinkoutput –const-fid family_id<br>plink –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf –allow-extra-chr –recode –vcf-half-call ‘haploid’ –out eee<br>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf –min-alleles 2 –max-alleles 2 –recode –out Vitis_all_sample</p>
<p>selscan –nsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf  –out nsl<br>selscan –ehh  –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.15use.vcf  –map &#x2F;home&#x2F;l&#x2F;data&#x2F;ehh&#x2F;ped&#x2F;output.map  –out ehh</p>
<p>sed -e ‘&#x2F;nan&#x2F;d’ tajimad_100.Tajima.D &gt; one.tajimad_100.Tajima.D &amp;&amp; sort -nr -k4 one.tajimad_100.Tajima.D &gt; sort.tajimaD</p>
<p>java -jar snpEff.jar databases |grep ‘Ovis’ |less -S<br>java -Xmx4g -jar snpEff.jar -o gatk -v &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf &gt; selected.ann.vcf<br>java -jar snpEff.jar -v Oar_v4.0 &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;0.85.vcf.recode.vcf &gt; &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;snpeff15.vcf<br>java -jar snpEff.jar build -gff3 -c snpEff.config -v oar_4 -d -noCheckCds -noCheckProtein<br>java -Xmx8g -jar snpEff.jar download -c snpEff.config -v Oar_v4.0</p>
<p>java -jar snpEff.jar build -c snpEff.config -gff3 -v Oar_v4.0 -d -noCheckCds -noCheckProtein<br>java -Xmx8g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;snpEff&#x2F;snpEff.jar ann -v oar_4 0.8.recode.vcf &gt; selected.ann.vcf<br>java -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;snpEff&#x2F;snpEff.jar ann oar_4 0.8.vcf.recode.vcf &gt; ann0.8new.vcf -csvStats snp.csv -stats snp.html</p>
<p>snpEff_v5_1_Ovis4.zip<br>java -jar snpEff.jar build -gtf22 -v Oar_v4.0<br>GCF_000298735.2_Oar_v4.0_genomic<br>java -jar snpEff.jar databases | grep -i Ovis_aries</p>
<p>awk ‘$3 &#x3D; “vcftools –vcf &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;snpeff.vcf –chr”” “ $1” “ “–from-bp”” “$2” “ “–to-bp”” “$2+100000” “ “–recode –recode-INFO-all –out”” “$1”_”$2 {print $3}’ &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;28top.site &gt; &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;note&#x2F;vcf.sh<br>sh vcf.sh<br>grep -v “int” *.recode.vcf|grep -v “synonymous_variant” &gt; non_int_synon.vcf<br>提取SNP位点所对应的基因ID信息</p>
<p>awk ‘{print $8}’ non_int_synon.vcf|grep -o ‘VJ01Gene[0-9][0-9][0-9][0-9][0-9]’ &gt; geneID.txt<br>awk ‘{print $8}’ non_int_synon.vcf &gt; geneID.txt</p>
<p>去重复</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">sort</span> -n geneID.txt | <span class="built_in">uniq</span> &gt; clean_geneID.txt</span><br></pre></td></tr></table></figure>

<p>将物种的基因ID与GO数据库的ID对应</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk <span class="string">&#x27;ARGIND==1&#123;a[$1]=$0&#125; ARGIND==2 &amp;&amp; ($1 in a) &#123;print $0&#125;&#x27;</span> clean_geneID.txt annot.go.tab &gt; geneID_goID.txt</span><br></pre></td></tr></table></figure>

<p>提取GO ID</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">awk <span class="string">&#x27;&#123;print $2&#125;&#x27;</span> geneID_goID.txt &gt; goID.txt</span><br><span class="line"></span><br><span class="line">awk <span class="string">&#x27;$3 = &quot;vcftools --vcf snpeff.vcf --chr&quot;&quot; &quot; $1&quot; &quot; &quot;--from-bp&quot;&quot; &quot;$2&quot; &quot; &quot;--to-bp&quot;&quot; &quot;$2+100000&quot; &quot; &quot;--recode --recode-INFO-all --out&quot;&quot; &quot;$1&quot;_&quot;$2 &#123;print $3&#125;&#x27;</span> 28top.site &gt; vcf.sh</span><br></pre></td></tr></table></figure>



<p>vcftools  –vcf 0.2change.vcf –plink –out 0.2cmiss</p>
<p>vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –IMPUTE</p>
<p>java -Xmx8g -jar &#x2F;home&#x2F;l&#x2F;anaconda3&#x2F;pkgs&#x2F;gatk-3.8-hdfd78af_11&#x2F;opt&#x2F;gatk-3.8&#x2F;GenomeAnalysisTk.jar&#x2F;GenomeAnalysisTK.jar -T VariantsToBinaryPed -R &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;data&#x2F;Oar_v4.0&#x2F;sequences.fa -V &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –metaData sampleID.fam -mgq 0 -bed phasing.bed -bim file.bim  -fam phasing.fam　<br>&#x2F;home&#x2F;l&#x2F;biosoft&#x2F;jdk-1.8&#x2F;jdk1.8.0_371&#x2F;jre&#x2F;bin&#x2F;java -Xmx8g -jar GenomeAnalysisTk.jar-T VariantsToBinaryPed -R &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;data&#x2F;Oar_v4.0&#x2F;sequences.fa -V &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –metaData sampleID.fam -mgq 0 -bed phasing.bed -bim file.bim  -fam phasing.fam　<br>selscan –ehh  –vcf 0.2change.vcf –map 0.2cmiss.map –out output<br>vcftools –vcf 0.2change.vcf –IMPUTE</p>
<p>#for循环计算fst，pi</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> &#123;1..31&#125;;<span class="keyword">do</span></span><br><span class="line">   <span class="comment">#根据ID，从vcf文件中提取每个亚群的信息</span></span><br><span class="line">   &#123;vcftools --vcf Filter.snp.vcf --keep <span class="variable">$&#123;i&#125;</span>-population.txt --recode -- </span><br><span class="line">   recode-INFO-all --out p<span class="variable">$&#123;i&#125;</span>&#125; &amp;</span><br><span class="line">   <span class="comment">#根据提取的信息计算每个亚群的PI值</span></span><br><span class="line">   &#123;vcftools --vcf p<span class="variable">$&#123;i&#125;</span>.recode.vcf --out q<span class="variable">$&#123;i&#125;</span>_pi_500kb --window-pi </span><br><span class="line">   100000 --window-pi-step 50000&#125; &amp;</span><br><span class="line">   <span class="comment">#根据提取的信息计算每个亚群的TajimaD值</span></span><br><span class="line">   &#123;vcftools --vcf p<span class="variable">$&#123;i&#125;</span>.recode.vcf --out q<span class="variable">$&#123;i&#125;</span>_100kb.TajimaD --TajimaD </span><br><span class="line">   100000&#125; &amp;</span><br><span class="line">   <span class="keyword">done</span></span><br></pre></td></tr></table></figure>



<p>#首先对vcf文件进行压缩</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bcftools view miss0.8.recode.vcf -Oz -o phasing</span><br><span class="line">bcftools view miss0.8.recode.vcf -Oz -o sample.vcf.gz</span><br></pre></td></tr></table></figure>

<p>#然后建立索引</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bcftools index Vitis_all_sample.recode.vcf.gz</span><br></pre></td></tr></table></figure>

<p>#最后根据群体分类表提取子vcf文件</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">bcftools view -S pop.list Vitis_all_sample.recode.vcf.gz &gt; new.vcf</span><br><span class="line"><span class="comment">#替换染色体号</span></span><br><span class="line">sed -e <span class="string">&#x27;s/NC_019459.2/2/g&#x27;</span> -e <span class="string">&#x27;s/NC_019458.2/1/g&#x27;</span> -e <span class="string">&#x27;s/NC_019460.2/3/g&#x27;</span> -e <span class="string">&#x27;s/NC_019461.2/4/g&#x27;</span> -e <span class="string">&#x27;s/NC_019462.2/5/g&#x27;</span> -e <span class="string">&#x27;s/NC_019463.2/6/g&#x27;</span> -e <span class="string">&#x27;s/NC_019464.2/7/g&#x27;</span> -e <span class="string">&#x27;s/NC_019465.2/8/g&#x27;</span> -e <span class="string">&#x27;s/NC_019466.2/9/g&#x27;</span> -e <span class="string">&#x27;s/NC_019467.2/10/g&#x27;</span> -e <span class="string">&#x27;s/NC_019468.2/11/g&#x27;</span> -e <span class="string">&#x27;s/NC_019469.2/12/g&#x27;</span> -e <span class="string">&#x27;s/NC_019470.2/13/g&#x27;</span> -e <span class="string">&#x27;s/NC_019471.2/14/g&#x27;</span> -e <span class="string">&#x27;s/NC_019472.2/15/g&#x27;</span> -e <span class="string">&#x27;s/NC_019473.2/16/g&#x27;</span> -e <span class="string">&#x27;s/NC_019474.2/17/g&#x27;</span> -e <span class="string">&#x27;s/NC_019475.2/18/g&#x27;</span> -e <span class="string">&#x27;s/NC_019476.2/19/g&#x27;</span> -e <span class="string">&#x27;s/NC_019477.2/20/g&#x27;</span>  -e <span class="string">&#x27;s/NC_019478.2/21/g&#x27;</span> -e <span class="string">&#x27;s/NC_019479.2/22/g&#x27;</span> -e <span class="string">&#x27;s/NC_019480.2/23/g&#x27;</span> -e <span class="string">&#x27;s/NC_019481.2/24/g&#x27;</span> -e <span class="string">&#x27;s/NC_019482.2/25/g&#x27;</span> -e <span class="string">&#x27;s/NC_019483.2/26/g&#x27;</span> -e <span class="string">&#x27;s/NC_019484.2/x/g&#x27;</span> -e <span class="string">&#x27;s/NC_001941.1/mt/g&#x27;</span> /home/l/data/vcf/miss0.8.recode.vcf &gt; 0.2change.vcf</span><br></pre></td></tr></table></figure>

<p>conda create -n java1.8 java&#x3D;1.8</p>
<p>#用plink过滤低频位点</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --vcf miss0.8.recode.vcf --maf 0.05 --geno 0.2 --recode vcf-iid -out 0.8-maf0.05 --allow-extra-chr</span><br></pre></td></tr></table></figure>

<p>#用plink筛选连锁区</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink --vcf 0.8-maf0.05.vcf --indep-pairwise 100 50 0.2 -out 0.8-maf0.05-LD --allow-extra-chr --make-bed</span><br></pre></td></tr></table></figure>

<p>#用plink找到强连锁区</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plink --noweb --bfile 0.8-maf0.05-LD --blocks no-pheno-req --allow-extra-chr</span><br><span class="line"></span><br><span class="line">plink --vcf miss0.8.recode.vcf --maf 0.05 --geno 0.2 --recode vcf-iid -out 0.8-maf0.05 --allow-extra-chr &amp;&amp; plink --vcf 0.8-maf0.05.vcf --indep-pairwise 100 50 0.2 -out 0.8-maf0.05-LD --allow-extra-chr --make-bed &amp;&amp; plink --noweb --bfile 0.8-maf0.05-LD --blocks no-pheno-req --allow-extra-chr</span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure>





<p>annotation_row &#x3D; data.frame(chr&#x3D;factor(c(rep(“1”,916),rep(“2”,892),rep(“3”,760),  rep(“4”,399),rep(“5”,358),rep(“6”,444),rep(“7”,317),rep(“8”,296),rep(“9”,272),rep(“10”,347),rep(“11”,256),rep(“12”,256),rep(“13”,305),rep(“14”,162),rep(“15”,264),rep(“16”,218),rep(“17”,237),rep(“18”,225),rep(“19”,213),rep(“20”,162),rep(“21”,144),rep(“22”,175),rep(“23”,198),rep(“24”,102),rep(“25”,142),rep(“26”,130),rep(“x”,259))))<br>row.names(annotation_row) &#x3D; colnames(data1)<br>c(916,1808,2568,2570,2968,3326,3770,4087,4383,4655,5002,5258,5514,5819,5981,6245,6463,6700,6925,7138,7300,7444,7619,7817,7919,8061,8191,8450)</p>
<p>perl &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;annovar&#x2F;retrieve_seq_from_fasta.pl –format refGene –seqfile genome.fna genome_refGene.txt –out genome_refGeneMrna.fa<br>perl &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;annovar&#x2F;convert2annovar.pl -format vcf4old  &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf &gt; 0.8.avinput</p>
<p>gtfToGenePred -genePredExt -ignoreGroupsWithoutExons ovis_replace.gtf ovis4_refGene.txt<br>perl &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;annovar&#x2F;annotate_variation.pl -dbtype refGene -out ovis_ano -buildver genome &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;annovar&#x2F;ovis&#x2F;0.8.avinput .&#x2F;genome</p>
<p>perl &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;annovar&#x2F;retrieve_seq_from_fasta.pl –format refGene –seqfile genome.fna genome_refGene.txt –out ovis.fa</p>
<p>cat top1.txt top2.txt top3.txt top4.txt top5.txt top6.txt top7.txt top8.txt top9.txt top10.txt top11.txt top12.txt top13.txt top14.txt top15.txt top16.txt top17.txt top18.txt top20.txt top21.txt top22.txt top23.txt top24.txt top25.txt top26.txt top27.txt top28.txt top29.txt top30.txt top31.txt&gt; topall.txt<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;25) print }’ pass&#x3D;1 topall.txt pass&#x3D;2 topall.txt &gt; topall25.site<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;20) print }’ pass&#x3D;1 topall.txt pass&#x3D;2 topall.txt &gt; topall20.site<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;15) print }’ pass&#x3D;1 topall.txt pass&#x3D;2 topall.txt &gt; topall15.site</p>
<p>for i in {1..31}; do sort -rn -k 6 &#x2F;home&#x2F;l&#x2F;data&#x2F;fst_window&#x2F;0.2&#x2F;50-25&#x2F;top$i.fst &gt;top$i.txt ; done</p>
<p>java  -Xmx8g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;snpEff&#x2F;snpEff.jar  nthreads&#x3D;6  ann -v Oar_v4.0 &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf &gt; ann.vcf<br>-xss5m</p>
<p>library(ggplot2)<br>mydata&lt;-read.table(“TajimaD500K.Tajima.D”,header &#x3D; T) #其他文件只需要换一下名字就好<br>mydata1&lt;-na.omit(mydata)<br>ggplot(mydata1, aes(x&#x3D;BIN_START&#x2F;1000000,y&#x3D;TajimaD,group&#x3D;factor(CHROM),colour&#x3D;CHROM))+geom_line()+facet_wrap(mydata1$CHROM)+xlab(“Physical distance(Mb)”)+ylab(“Tajima’s D”)+theme(legend.position &#x3D; “none”)<br>a&lt;-mydata1[sample(nrow(mydata1), 1000), ] #随机选取其中的1000行数据，这个Hardy-Weinberg平衡文件需要的，因为结果太多，无法利用ggplot绘出全部结果</p>
<p>for i in {1..31}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’  w150_75hu_$i.fst.windowed.weir.fst &gt; a$i.txt;done<br>cat a1.txt a2.txt a3.txt a4.txt a5.txt a6.txt a7.txt a8.txt a9.txt a10.txt a11.txt a12.txt a13.txt a14.txt a15.txt a16.txt a17.txt a18.txt a19.txt a20.txt a21.txt a22.txt a23.txt a24.txt a25.txt a26.txt a27.txt a28.txt a29.txt a30.txt a31.txt &gt; mergea.txt<br>sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’  topall20.site &gt; topall20.txt<br>sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;27&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’  oar.fasta &gt; coar.fasta</p>
<p>awk ‘$3 &#x3D; “vcftools –vcf ann.vcf –chr”” “ $1” “ “–from-bp”” “$2” “ “–to-bp”” “$3” “ “–recode –recode-INFO-all –out”” “$1”_”$2 {print $3}’ m15.txt &gt; vcf.sh<br>grep -v “int” ann.vcf|grep -v “synonymous_variant” &gt; non_int_synon.vcf</p>
<p>vcf转hap<br>添加 -sortPositions 参数，以便在转换之前进行位点排序</p>
<p>run_pipeline.pl -Xms40g -Xmx40g  -vcf gt_group19.vcf -sortPositions -export hapd -exportType HapmapDiploid<br>run_pipeline.pl -Xms6g -Xmx60g  -vcf gt_group1.vcf -sortPositions -export plink -exportType Plink<br>run_pipeline.pl -Xms6G -Xmx60G -fork1 -sortPositions -vcf use.vcf -export hapmap  -exportType Hapmap<br>run_pipeline.pl -Xms8g -Xmx80g  -vcf gt_group19.vcf -sortPositions -export hap -exportType Hapmap<br>vcftools –vcf gt_group1.vcf –IMPUTE<br>bash &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;anno&#x2F;100kb&#x2F;get_snp.sh &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;anno&#x2F;100kb&#x2F;p25.txt &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;anno&#x2F;ann.vcf </p>
<p>vcftools –gzvcf doporchr6.vcf.gz –IMPUTE<br>java -Xms80g -Xmx80g -jar   &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;8 gt&#x3D;group19.vcf out&#x3D;gt_group19</p>
<p>for i in {1..5}; do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done<br>for i in {6..10};  do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done<br>for i in {11..15};  do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done<br>for i in {16..20}; do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done<br>for i in {21..25}; do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done<br>for i in {26..31};  do vcftools –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf –recode –recode-INFO-all –stdout  –indv group$i  &gt; group$i.vcf;done</p>
<p>.&#x2F;VCF_add_id.pl cid.vcf aid.vcf<br>bcftools view cid.vcf -Oz -o ruse.vcf.gz<br>bcftools index use.vcf.gz</p>
<p>for i in {1..27}; do vcftools –vcf cid.vcf –chr $i –recode –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr$i ; done<br>vcftools –vcf cid.vcf –chr 3 –recode –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr3<br>vcftools –vcf cid.vcf –chr 6 –recode –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr6<br>vcftools –vcf cid.vcf –chr 10 –recode –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr10<br>vcftools –vcf cid.vcf –chr 13 –recode –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr13<br>bcftools view cid.vcf -Oz -o caid.vcf.gz</p>
<p>for i in {1..8}; do bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;grouph&#x2F;group$i.txt use.vcf.gz &gt; phasinggroup$i.vcf ; done<br>for i in {9..16}; do bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;grouph&#x2F;group$i.txt use.vcf.gz &gt; phasinggroup$i.vcf ; done<br>for i in {17..24}; do bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;grouph&#x2F;group$i.txt use.vcf.gz&gt; phasinggroup$i.vcf ; done<br>for i in {24..31}; do bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;grouph&#x2F;group$i.txt use.vcf.gz &gt; phasinggroup$i.vcf ; done</p>
<p>nohup bcftools view selscan.vcf.recode.vcf -Oz -o use.vcf.gz &amp;<br>bcftools index use.vcf.gz<br>bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;grouph&#x2F;feizhou.txt use.vcf.gz &gt; fz.vcf<br>bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;grouph&#x2F;yinjin.txt use.vcf.gz &gt; yj.vcf<br>bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;grouph&#x2F;difang.txt use.vcf.gz &gt; df.vcf</p>
<p>for i in {6..10}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done<br>for i in {11..15}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done<br>for i in {16..20}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done<br>for i in {21..25}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done<br>for i in {26..31}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done<br>for i in {1..5}; do bcftools view &#x2F;home&#x2F;l&#x2F;data&#x2F;vcf&#x2F;miss0.8.recode.vcf -S &#x2F;home&#x2F;l&#x2F;data&#x2F;grouph&#x2F;group$i.txt -o group$i.vcf;done</p>
<p>for i in {26..31}; do java -Xms80g -Xmx80g -jar   &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;12 gt&#x3D;group$i.vcf out&#x3D;gt_group$i ;done</p>
<p>vcftools –vcf ann.vcf –IMPUTE</p>
<p>selscan –ehh  –hap hu.hap –map nhuchr_6.map –out selscanhu6<br>ihsbin –hap hu.hap –map new.map –out ihsbinhu61<br>ehhbin –hap out.impute.hap –map plink.plk.map –locus 54 &gt; 443454_EHH.txt<br>selscan –xpnsl –vcf gt_group1.vcf  –vcf-ref gt_group19.vcf –out 1_19<br>selscan –xpehh –hap hu.hap –ref dopor.hap –map new.map –out xpehhhudopor61selscan<br>xpehhbin –hapA &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;hu.hap –hapB &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;yy.hap –map hu.map –out xpehhu_yy<br>for i in {1..26}; do selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;yychr$i.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_$i.map –out hu_yy$i –threads 14 ;done<br>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchrx.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;yychrx.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_x.map –out hu_yyx –threads 12<br>cat hu_yy1.xpehh.out hu_yy2.xpehh.out  hu_yy3.xpehh.out  hu_yy4.xpehh.out  hu_yy5.xpehh.out  hu_yy6.xpehh.out  hu_yy7.xpehh.out  hu_yy8.xpehh.out  hu_yy9.xpehh.out hu_yy10.xpehh.out hu_yy11.xpehh.out hu_yy12.xpehh.out hu_yy13.xpehh.out hu_yy14.xpehh.out  hu_yy15.xpehh.out  hu_yy16.xpehh.out  hu_yy17.xpehh.out  hu_yy18.xpehh.out  hu_yy19.xpehh.out  hu_yy20.xpehh.out  hu_yy21.xpehh.out  hu_yy22.xpehh.out  hu_yy23.xpehh.out  hu_yy24.xpehh.out  hu_yy25.xpehh.out  hu_yy26.xpehh.out  hu_yyx.xpehh.out &gt; xpehh.merge<br>selscan–vcf”${i}.vcf.gz”–vcf-ref”..&#x2F;w&#x2F;${i}.vcf.gz”–map”..&#x2F;map&#x2F;${i}”–threads 6 –out$i<br>selscan –ehh  –vcf gt_group1.vcf –map 1.map –out selscanehh1<br>for i in {1..8}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ group$i.vcf &gt; cgroup$i.vcf ;done<br>for i in {9..16}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ group$i.vcf &gt; cgroup$i.vcf ;done<br>for i in {17..24}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ group$i.vcf &gt; cgroup$i.vcf ;done<br>for i in {25..31}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ group$i.vcf &gt; cgroup$i.vcf ;done<br>for i in {1..31}; do sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ group$i.vcf &gt; cgroup$i.vcf ;done</p>
<p>sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;27&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’  0.8.vcf.recode.vcf &gt; cid.vcf</p>
<p>nohup vcftools –vcf aid.vcf –min-alleles 2 –max-alleles 2 –recode –out selscan.vcf &amp;<br>cut -f 1,3,2 caid.vcf &gt; a.txt &amp;&amp; sed -i ‘&#x2F;^#&#x2F;d’ a.txt &gt; b.txt<br>awk ‘{$3&#x3D;$1”<em>“$2{print $0}’; print}’ a.txt &gt; busy.map<br>awk ‘{$3&#x3D;$1”</em>“$2; print}’ a.txt &gt; busy.map<br>awk ‘{ print $1” “$3” “$2*1” “$2}’ busy.map &gt; use.map</p>
<p>for i in {1..27}; do vcftools –vcf  –chr $i –recode –out chr$i ; done</p>
<p>for i in {7..31}; do mkdir $i ; done</p>
<p>for i in {1..27}; do java -Xms50g -Xmx50g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf map&#x3D;&#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;map&#x2F;chr_$i.map nthreads&#x3D;30  out&#x3D;df_chr$i ;done<br>vcftools –vcf chr1.recode.vcf –IMPUTE<br>for i in {1..27}; do java -Xms200g -Xmx200g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf  nthreads&#x3D;120  out&#x3D;18_chr$i ;done<br>for i in {1..27}; do java -Xms200g -Xmx200g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf  nthreads&#x3D;120  out&#x3D;19_chr$i ;done<br>for i in {1..27}; do java -Xms200g -Xmx200g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf  nthreads&#x3D;120  out&#x3D;20_chr$i ;done</p>
<p>for i in {1..27}; do selscan –xpnsl –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;yy&#x2F;yy_chr$i.vcf.gz  –out hu_yy$i –threads 63 ;done</p>
<p>for i in {1..27}; do norm –xpnsl –files &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpnsl&#x2F;fz&#x2F;hu_fz$i.xpnsl.out –bp-win –winsize 150000  ;done<br>for i in {1..27}; do norm –xpnsl –files &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpnsl&#x2F;df&#x2F;hu_df$i.xpnsl.out –bp-win –winsize 150000  ;done</p>
<p>hu_df1.xpehh.out.norm.150kb.windows<br>cat hu* &gt; black.xpehh<br>hu_yjx.xpehh.out.norm<br>cat *.out.norm &gt; black.xpehh</p>
<p>norm –xpnsl –files hu_yjx.xpehh.out –bp-win –winsize 150000<br>for o in {1..8};do for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;$o&#x2F;$o_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;$o&#x2F;hu_$ochr$i –threads 30 ;done ;done<br>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;yy&#x2F;yy_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;1&#x2F;hu_1chr$i –threads 120 ;done<br>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;7&#x2F;7_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;7&#x2F;hu_7chr$i –threads 30 ;done<br>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;8&#x2F;8_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;8&#x2F;hu_8chr$i –threads 30 ;done<br>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;9&#x2F;9_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;9&#x2F;hu_9chr$i –threads 30 ;done</p>
<p>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;hu&#x2F;hu_chr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;VCF_add_id&#x2F;30&#x2F;30_chr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;map&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;30&#x2F;hu_30chr$i –threads 30 ;done</p>
<p>for i in {1..27}; do java -Xms200g -Xmx200g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf  nthreads&#x3D;120  out&#x3D;29_chr$i ;done<br>for i in {1..27}; do java -Xms200g -Xmx200g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf  nthreads&#x3D;120  out&#x3D;30_chr$i ;done</p>
<p>for i in {4..5}; do cd &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;xpehh&#x2F;$i; norm –xpnsl –files *.out –bp-win –winsize 100000  ;done</p>
<p>java -Xms80g -Xmx80g -jar   &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;10 gt&#x3D;doporchr6.recode.vcf  out&#x3D;dopor6<br>awk ‘{$3&#x3D;$1.$2; print}’ id19.vcf &gt; cd1.map<br>python addid.py –input cgroup19.vcf –output id19.vcf<br>cat chu.vcf | awk -v OFS&#x3D;”\t” ‘&#x2F;^#&#x2F;|| $3&#x3D;$1”<em>“$2{print $0}’ &gt;idchu.vcf<br>map&#x3D;huchr_6.map<br>awk ‘{print &gt; “chr</em>“$1”.map”}’ hu.map<br>vcftools –vcf id29.vcf –chr 6 –recode –out doporchr6<br>vcftools –vcf huchr6.recode.vcf –plink –out hu6<br>awk ‘{ print $1” “$3” “$2”}’ file.map &gt; new.map<br>Vcftools –vcf huchr6.recode.vcf –min-alleles 2 –max-alleles 2 –recode –out Vitis_all_sample<br>cut -f 1,2,4 nhuchr_6.map &gt; dis.map<br>awk ‘{ print $1” “$3” “$2*0.0000014” “$2}’ id19.map &gt; hu.map<br>3632cm ——– 2,587,499,057          ——-0.0000014<br>bcftools view -S yinjin.txt Vitis_all_sample.recode.vcf.gz &gt; &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yj.vcf<br>sed  -e ‘s&#x2F;NC_019458.2&#x2F;chr1&#x2F;g’ -e ‘s&#x2F;NC_019459.2&#x2F;chr2&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;chr3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;chr4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;chr5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;chr6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;chr7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;chr8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;chr9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;chr10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;chr11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;chr12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;chr13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;chr14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;chr15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;chr16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;chr17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;chr18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;chr19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;chr20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;chr21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;chr22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;chr23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;chr24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;chr25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;chr26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;chrx&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;chrmt&#x2F;g’ group19.vcf&gt;chu.vcf<br>java -Xms80g -Xmx80g -jar   &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;10 gt&#x3D;cyy.vcf map&#x3D;hu.map out&#x3D;pyy<br>for i in {1..27};do vcftools –vcf hu.vcf –chr $i –recode –out chr$i;done<br>vcftools –vcf hu.vcf –chr $i –recode –out chr$i<br>sed -e ‘s&#x2F;NC_019459.2&#x2F;2&#x2F;g’ -e ‘s&#x2F;NC_019458.2&#x2F;1&#x2F;g’ -e ‘s&#x2F;NC_019460.2&#x2F;3&#x2F;g’ -e ‘s&#x2F;NC_019461.2&#x2F;4&#x2F;g’ -e ‘s&#x2F;NC_019462.2&#x2F;5&#x2F;g’ -e ‘s&#x2F;NC_019463.2&#x2F;6&#x2F;g’ -e ‘s&#x2F;NC_019464.2&#x2F;7&#x2F;g’ -e ‘s&#x2F;NC_019465.2&#x2F;8&#x2F;g’ -e ‘s&#x2F;NC_019466.2&#x2F;9&#x2F;g’ -e ‘s&#x2F;NC_019467.2&#x2F;10&#x2F;g’ -e ‘s&#x2F;NC_019468.2&#x2F;11&#x2F;g’ -e ‘s&#x2F;NC_019469.2&#x2F;12&#x2F;g’ -e ‘s&#x2F;NC_019470.2&#x2F;13&#x2F;g’ -e ‘s&#x2F;NC_019471.2&#x2F;14&#x2F;g’ -e ‘s&#x2F;NC_019472.2&#x2F;15&#x2F;g’ -e ‘s&#x2F;NC_019473.2&#x2F;16&#x2F;g’ -e ‘s&#x2F;NC_019474.2&#x2F;17&#x2F;g’ -e ‘s&#x2F;NC_019475.2&#x2F;18&#x2F;g’ -e ‘s&#x2F;NC_019476.2&#x2F;19&#x2F;g’ -e ‘s&#x2F;NC_019477.2&#x2F;20&#x2F;g’  -e ‘s&#x2F;NC_019478.2&#x2F;21&#x2F;g’ -e ‘s&#x2F;NC_019479.2&#x2F;22&#x2F;g’ -e ‘s&#x2F;NC_019480.2&#x2F;23&#x2F;g’ -e ‘s&#x2F;NC_019481.2&#x2F;24&#x2F;g’ -e ‘s&#x2F;NC_019482.2&#x2F;25&#x2F;g’ -e ‘s&#x2F;NC_019483.2&#x2F;26&#x2F;g’ -e ‘s&#x2F;NC_019484.2&#x2F;x&#x2F;g’ -e ‘s&#x2F;NC_001941.1&#x2F;mt&#x2F;g’ yj.vcf &gt; cyj.vcf</p>
<p>for i in {1..12};do java -Xms90g -Xmx90g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf map&#x3D;chr_$i.map nthreads&#x3D;6  out&#x3D;dfchr$i ;done<br>for i in {13..26};do java -Xms90g -Xmx90g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf map&#x3D;chr_$i.map nthreads&#x3D;6  out&#x3D;dfchr$i ;done<br>java -Xms90g -Xmx90g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chrx.recode.vcf map&#x3D;chr_x.map nthreads&#x3D;6  out&#x3D;dfchrx</p>
<p>nohup java -Xms200g -Xmx200g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;60 gt&#x3D;g2id.vcf map&#x3D;new.map out&#x3D;g2phasing &amp;<br>for i in {1..26};do java -Xms30g -Xmx30g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr$i.recode.vcf map&#x3D;chr_$i.map nthreads&#x3D;6  out&#x3D;huchr$i ;done<br>xpehhbin –hapA &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;phu.vcf –hapB &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;pyy.vcf –map hu.map –out xpehhu_yy<br>$selscan–vcf”${i}.vcf.gz”–vcf-ref”..&#x2F;w&#x2F;${i}.vcf.gz”–map”..&#x2F;map&#x2F;${i}”–threads 6 –out$i<br>java -Xms30g -Xmx30g -jar &#x2F;home&#x2F;l&#x2F;biosoft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chrx.recode.vcf map&#x3D;chr_x.map nthreads&#x3D;13  out&#x3D;fzchrx </p>
<p>for i in {4..5}; do selscan –xpnsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;df&#x2F;dfchr$i.vcf.gz  –out hu_df$i –threads 14 ;done<br>for i in {1..26}; do selscan –xpnsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yj&#x2F;yjchr$i.vcf.gz –out hu_yj$i –threads 14 ;done<br>for i in {1..26}; do selscan –xpnsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;fzchr$i.vcf.gz  –out hu_fz$i –threads 14 ;done<br>for i in {1..26}; do selscan –xpnsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;yychr$i.vcf.gz –out hu_yy$i –threads 14 ;done<br>selscan –xpnsl –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchrx.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;fzchrx.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_x.map –out hu_fzx –threads 14</p>
<p>#selscan计算XP-ehh<br>###selscan –xpehh –vcf  –vcf-ref  –map  –out<br>for i in {1..26};do selscan –xpehh –vcf  –vcf-ref  –map  –out ;done<br>#selscan计算XP-nSL<br>###selscan –xpnsl –vcf  –vcf-ref  –out<br>for i in {1..26};do selscan –xpnsl –vcf  –vcf-ref  –out ;done<br>#selscan计算ihs<br>selscan –ihs –vcf  –map  –out </p>
<p>#selscan子程序norm标准化<br>#–bp-win设置窗口<br>#To normalize iHS output across frequency bins:<br>.&#x2F;norm –ihs –files &lt;file1.ihs.out&gt; … &lt;fileN.ihs.out&gt;<br>#To normalize nSL output across frequency bins:<br>.&#x2F;norm –ihs –files &lt;file1.nsl.out&gt; … &lt;fileN.nsl.out&gt;<br>#To normalize iHS output and analyze non-overlapping windows of fixed bp for extreme iHS scores:<br>.&#x2F;norm –ihs –files &lt;file1.ihs.out&gt; … &lt;fileN.ihs.out&gt; –bp-win<br>#To normalize XP-EHH output across frequency bins:<br>.&#x2F;norm –xpehh –files &lt;file1.xpehh.out&gt; … &lt;fileN.xpehh.out&gt;<br>#To normalize XP-EHH output and analyze non-overlapping windows of fixed bp for extreme XP-EHH scores:<br>.&#x2F;norm –xpehh –files &lt;file1.xpehh.out&gt; … &lt;fileN.xpehh.out&gt; –bp-win</p>
<p>vcftools –vcf cfz.vcf –min-alleles 2 –max-alleles 2 –recode –out cfz2.vcf<br>awk ‘{for(i &#x3D; 1; i &lt;&#x3D; NF; i++) if(i &gt;&#x3D; 10 &amp;&amp; i &lt;&#x3D; 34 &amp;&amp; $i &#x3D;&#x3D; “.”) {$i &#x3D; “.&#x2F;.”} {print $0}}’ chr5.recode.vcf | tr -s ‘ ‘ ‘\t’ &gt; chr5.vcf</p>
<p>cut -f 1,3,2 cfz2.vcf.recode.vcf &gt; cfz.map<br>sed -i ‘&#x2F;^#&#x2F;d’ cfz.map</p>
<p>for i in {1..26}; do awk ‘{$3&#x3D;$1”<em>“$2;print $0}’ chr$i.recode.vcf &gt; idhuchr$i.vcf ; done<br>for i in {1..26}; do awk ‘{$3&#x3D;$1”</em>“$2;print $0}’ chr$i.recode.vcf &gt; idhuchr$i.vcf ; done<br>for i in {1..26}; do awk ‘{$3&#x3D;$1”<em>“$2;print $0}’ chr$i.recode.vcf &gt; idhuchr$i.vcf ; done<br>for i in {1..26}; do awk ‘{$3&#x3D;$1”</em>“$2;print $0}’ chr$i.recode.vcf &gt; idhuchr$i.vcf ; done<br>for i in {1..26}; do awk ‘{$3&#x3D;$1”_”$2;print $0}’ chr$i.recode.vcf &gt; idhuchr$i.vcf ; done</p>
<p>awk ‘{$3&#x3D;$1”_”$2;print $0}’ c0.2.vcf &gt; iduse.vcf</p>
<p>awk ‘{ print $1” “$3” “$2*0.0000014” “$2}’ cfz1.map &gt; ufz.map<br>awk ‘{print &gt; “chr_”$1”.map”}’ ufz.map</p>
<p>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchrx.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;df&#x2F;dfchrx.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_x.map –out hu_dfx –threads 14<br>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchrx.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yj&#x2F;yjchrx.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_x.map –out hu_yjx –threads 14<br>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchrx.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;yy&#x2F;yychrx.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;chr_x.map –out hu_yyx –threads 14<br>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr4.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;rfzchr4.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;fzmap&#x2F;chr_4.map –out hu_fz4 –threads 14<br>selscan –xpehh –vcf &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;hu&#x2F;huchr5.vcf.gz –vcf-ref &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;rfzchr5.vcf.gz –map &#x2F;home&#x2F;l&#x2F;data&#x2F;gz&#x2F;group&#x2F;fz&#x2F;fzmap&#x2F;chr_5.map –out hu_fz5 –threads 14</p>
<p>vcftools –vcf 855fz.vcf –chr 4 –recode –out chr4<br>vcftools –vcf 855fz.vcf –chr 5 –recode –out chr5</p>
<p>chown liuxiao2022 -R &#x2F;mnt&#x2F;d&#x2F;liuxiao2022<br>useradd  -g root  liuxiao<br>chown zhuang -R &#x2F;home&#x2F;zhuang<br>usermod -d &#x2F;mnt&#x2F;d&#x2F;liuxiao2022 liuxiao<br>grep liuxiao &#x2F;etc&#x2F;passwd<br>mount -o rw &#x2F;dev&#x2F;sdc1 &#x2F;media&#x2F;myharddisk</p>
<p>zcat ann.vcf | awk -v OFS&#x3D;”\t” ‘&#x2F;^#&#x2F;|| $3&#x3D;$1”_”$2{print $0}’ &gt; finally.vcf</p>
<p>perl VCF_add_id.pl ann.vcf annid.vcf</p>
<p>java -Xms150g -Xmx150g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;yj.vcf map&#x3D;use.map nthreads&#x3D;30  out&#x3D;yj<br>java -Xms150g -Xmx150g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;df.vcf map&#x3D;use.map nthreads&#x3D;30 out&#x3D;df<br>java -Xms150g -Xmx150g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;fz.vcf map&#x3D;use.map nthreads&#x3D;30  out&#x3D;fz<br>java -Xms150g -Xmx150g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar gt&#x3D;yj.vcf map&#x3D;chr_$i.map nthreads&#x3D;6  out&#x3D;dfchr$i </p>
<p>vcftools –vcf huscore.vcf –extract-FORMAT-info GT –out specific_snp<br>cat specific_snp.GT.FORMAT | cut -f 1,2,308,309,310,311,312,313,314,315,316,317,318 &gt; barcode.GT.FORMAT</p>
<p>bcftools view &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;ann0.8new.vcf -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;grouph&#x2F;group19.txt &gt; husite.vcf<br>bcftools view intergenic_region.vcf -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;grouph&#x2F;group19.txt &gt; bara.vcf</p>
<p>$ echo -e “##fileformat&#x3D;VCFv4.2\n##contig&#x3D;&lt;ID&#x3D;1,length&#x3D;249250621,assembly&#x3D;b37&gt;\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n1\t1\t.\tA\tC\t.\t.\t.” | &#x2F;usr&#x2F;bin&#x2F;bcftools +fill-AN-AC –no-version<br>bcftools concat concat-a.vcf concat-b.vcf -o combine_a_b_samesample_diffsites_bcftools.vcf<br>vcf-concat concat-a.vcf concat-b.vcf &gt; combine_a_b_samesample_diffsites_vcftools.vc</p>
<p>bash &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;get_snp.sh &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;site&#x2F;mcsite15.txt husite.vcf.gz<br>bcftools concat concat-a.vcf concat-b.vcf -o combine_a_b_samesample_diffsites_bcftools.vcf<br>bcftools concat *.vcf  -o combine_a_b_samesample_diffsites_bcftools.vcf</p>
<p>bcftools consensus -f reference.fasta extracted_loci.vcf &gt; extracted_loci.fasta</p>
<p>bcftools view specific_position.vcf -Oz -o bar.vcf.gz<br>bcftools index bar.vcf.gz<br>bcftools merge *.vcf.gz  -o merged.vcf</p>
<p>bcftools filter 1000Genomes.vcf.gz –regions &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;site&#x2F;mcsite15.txt &gt; vcf</p>
<p>安装必要的工具：</p>
<p>VCFtools：一套用于处理 VCF 文件的工具。<br>bcftools：一套用于变异检测和操作 VCF 和 BCF 文件的实用程序。<br>seqtk：一个用于处理 FASTA 或 FASTQ 格式序列的快速轻量级工具。<br>您可以使用诸如 apt 或 conda 的包管理器安装这些工具，或从源代码编译它们。</p>
<p>确定感兴趣的位点：</p>
<p>您应该有一个要从 VCF 文件中提取的位点列表。这可以是一个基因组坐标列表或特定标记名称。<br>使用 VCFtools 或 bcftools 提取感兴趣的位点：</p>
<p>使用 VCFtools：<br>vcftools –vcf husite.vcf –positions merged15.txt –recode –recode-INFO-all –out specific_position.vcf<br>将 input.vcf 替换为您的 VCF 文件，将 loci.txt 替换为包含感兴趣位点的文件，每行一个。</p>
<p>使用 bcftools：</p>
<p>bcftools view -R site6.txt chr6.vcf.gz &gt; out specific_position.vcf<br>将 input.vcf 替换为您的 VCF 文件，将 loci.txt 替换为包含感兴趣位点的文件，格式为 CHROM:POS。</p>
<p>将提取的 VCF 转换为 FASTA 格式：</p>
<p>您可以使用 bcftools 将 VCF 文件转换为一致性 FASTA 文件：<br>bcftools consensus -f oar.fasta site5.vcf.gz &gt; site5.fasta<br>将 reference.fasta 替换为 FASTA 格式的参考基因组。<br>生成 DNA 条形码：</p>
<p>您可以使用 seqtk 从一致性 FASTA 文件中提取特定区域并创建 DNA 条形码：<br>seqtk subseq site5.fasta site5.txt &gt; barcode5.fasta<br>将 regions.txt 替换为包含感兴趣区域的文件，格式为 CHROM:START-END。<br>此流程从 VCF 文件中提取特定位点，将提取的位点转换为一致性 FASTA 文件，并使用确定的区域生成 DNA 条形码。</p>
<p>conda create -n qiime2 –file<br>conda activate qiime2<br>wget <a target="_blank" rel="noopener" href="https://data.qiime2.org/distro/core/qiime2-2023.5-py38-linux-conda.yml">https://data.qiime2.org/distro/core/qiime2-2023.5-py38-linux-conda.yml</a><br>conda env create -n qiime2-2023.5 –file qiime2-2023.5-py38-linux-conda.yml</p>
<p>for i in {1..31} ; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr13.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;13&#x2F;hu_$i ;done<br>vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;aid.vcf –chr 13 –recode –out chr13</p>
<p>#将合并相邻窗口后overlap窗口取交集<br>cat top1.txt top2.txt top3.txt top4.txt top5.txt top6.txt top7.txt top8.txt top9.txt top10.txt top11.txt top12.txt top13.txt top14.txt top15.txt top16.txt top17.txt top18.txt top20.txt top21.txt top22.txt top23.txt top24.txt top25.txt top26.txt top27.txt top28.txt top29.txt top30.txt top31.txt&gt; topall.txt<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;1) print }’ pass&#x3D;1 lmtop.txt pass&#x3D;2 lmtop.txt &gt; mt.txt &amp;&amp; cut -f 1,2 mt.txt &gt; msite24.txt &amp;&amp; awk ‘!seen[$0]++’ msite24.txt &gt; site1.txt<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;2) print }’ pass&#x3D;1 lmtop.txt pass&#x3D;2 lmtop.txt &gt; mt.txt &amp;&amp; cut -f 1,2 mt.txt &gt; msite24.txt &amp;&amp; awk ‘!seen[$0]++’ msite24.txt &gt; site2.txt<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;3) print }’ pass&#x3D;1 lmtop.txt pass&#x3D;2 lmtop.txt &gt; mt.txt &amp;&amp; cut -f 1,2 mt.txt &gt; msite24.txt &amp;&amp; awk ‘!seen[$0]++’ msite24.txt &gt; site3.txt</p>
<p>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;10) print }’ pass&#x3D;1 mtop.txt pass&#x3D;2 mtop.txt &gt; mt.txt &amp;&amp; cut -f 1,2 mt.txt &gt; msite24.txt &amp;&amp; awk ‘!seen[$0]++’ msite24.txt &gt; site10.txt<br>awk ‘pass&#x3D;&#x3D;1 { count[$1,$2]++ } pass&#x3D;&#x3D;2 { if(count[$1,$2]&gt;5) print }’ pass&#x3D;1 mtop.txt pass&#x3D;2 mtop.txt &gt; mt.txt &amp;&amp; cut -f 1,2 mt.txt &gt; msite24.txt &amp;&amp; awk ‘!seen[$0]++’ msite24.txt &gt; site5.txt</p>
<p>cut -f 1,2 chr13topall15site.txt &gt; topall15site.txt<br>cut -f 1,2 chr13topall18site.txt &gt; topall18site.txt<br>cut -f 1,2 chr13topall24site.txt &gt; topall24site.txt<br>cut -f 1,2 chr13topall27site.txt &gt; topall27site.txt<br>cut -f 1,2 chr13topall28site.txt &gt; topall28site.txt</p>
<p>awk ‘!seen[$0]++’ topall15site.txt &gt; chr13topall15site.txt<br>awk ‘!seen[$0]++’ topall18site.txt &gt; chr13topall18site.txt<br>awk ‘!seen[$0]++’ topall24site.txt &gt; chr13topall24site.txt<br>awk ‘!seen[$0]++’ topall27site.txt &gt; chr13topall27site.txt<br>awk ‘!seen[$0]++’ topall28site.txt &gt; chr13topall28site.txt</p>
<p>cat chr13topall18site.txt chr10topall18site.txt chr8topall18site.txt chr6topall18site.txt chr3topall18site.txt &gt; merged18.txt</p>
<p>vcftools –vcf specific_position.vcf.recode.vcf –extract-FORMAT-info GT –out specific_snp<br>bcftools view specific_position.vcf.recode.vcf | awk ‘ {print $4, $5}’ &gt; atcg.txt<br>bcftools view chr6.recode.vcf | awk ‘$2 &gt;&#x3D; 29417592 &amp;&amp; $2 &lt;&#x3D; 29418207’<br>vcftools –vcf husite.vcf –chr NC_019463.2 –recode –out chr6</p>
<p>for i in {1..2};do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;aid.vcf  –chr $i –recode –out chr$i ; done</p>
<p>for i in {1..27} ;do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;chr&#x2F;chr$i.recode.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group1.txt  –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;fsthugroup1&#x2F;chr$i ; done</p>
<p>cd &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;fsthugroup31<br>cat <em>fst &gt; m.txt<br>awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; mmmm.txt &amp;&amp; head -n 100000 mmmm.txt &gt; top$i.txt<br>mv top</em> &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode</p>
<p>lmtop.txt</p>
<p>mv top* &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode</p>
<p>bcftools view -R site6.txt chr6.vcf.gz  -o<br>bcftools view &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;aid.vcf -Oz -o &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode&#x2F;bar.vcf.gz<br>bcftools index bar.vcf.gz<br>for i in {1..30} ; do bcftools view -R site$i.txt hu.vcf.gz &gt; site$i.vcf ; done<br>for i in {1..2} ; do bcftools view -R site$i.txt bar.vcf.gz &gt; site$i.vcf ; done<br>for i in {3..5} ; do bcftools view -R site$i.txt bar.vcf.gz &gt; site$i.vcf ; done<br>for i in {6..14} ; do bcftools view -R site$i.txt bar.vcf.gz &gt; site$i.vcf ; done<br>bcftools view -S &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt bar.vcf.gz -o hu.vcf</p>
<p>for i in {1..11};do cd &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;fsthugroup$i &amp;&amp; awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m$i.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; mmmm.txt &amp;&amp; head -n 40000 mmmm.txt &gt; &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode&#x2F;ptop$i.txt;done<br>for i in {12..31};do cd &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;fsthugroup$i &amp;&amp; awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; mmmm.txt &amp;&amp; head -n 40000 mmmm.txt &gt; &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode&#x2F;ptop$i.txt;done</p>
<p>cat ptop* &gt; mptop.txt<br>wc -l mptop.txt<br>sort -nr -k3 mptop.txt &gt; laji.txt<br>tail laji.txt<br>awk ‘{ print $1 “_” $2 “\t” $0 }’ sortmergetop0.15.txt &gt; maptop.txt &amp;&amp; cut -f 1,2,3,4 maptop.txt &gt; macptop.txt &amp;&amp; sort macptop.txt  | uniq -c &gt;  0.15.txt</p>
<p>bcftools view -R bigsitescore.txt hu.vcf.gz &gt; &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_snp&#x2F;barcode&#x2F;scoresite&#x2F;huscore.vcf<br>bcftools consensus -f coar.fasta huscore.vcf.gz &gt; huscore.fasta<br>seqtk subseq huscore.fasta bigsitescore.txt &gt; barcode.fasta</p>
<p>plink –vcf huscorephasing.vcf.vcf –allow-extra-chr –recode –vcf-half-call ‘haploid’ –chr-set 27 –out eee</p>
<p>sed -e ‘s&#x2F;1&#x2F;1&#x2F;A&#x2F;g’ -e ‘s&#x2F;0&#x2F;1&#x2F;M&#x2F;g’ -e ‘s&#x2F;.&#x2F;.&#x2F;N&#x2F;g’ -e ‘s&#x2F;0&#x2F;0&#x2F;A&#x2F;g’  specific_snp.GT.FORMAT &gt; CGT.vcf</p>
<p>##优先染色体</p>
<p>for i in {1..31}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 100000 –fst-window-step 50000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window&#x2F;fstchr1&#x2F;w100hu_$i.fst ;done<br>for i in {1..31}; do  awk ‘{if($6<del>&#x2F;e&#x2F;)$6&#x3D;0}1’ w100hu_$i.fst.windowed.weir.fst &gt; aw100hu_$i.fst.windowed.weir.fst &amp;&amp; awk ‘{if($6</del>&#x2F;nan&#x2F;)$6&#x3D;0}1’ aw100hu_$i.fst.windowed.weir.fst &gt; bw100hu_$i.fst.windowed.weir.fst &amp;&amp; sort -nr -k6 bw100hu_$i.fst.windowed.weir.fst &gt; cw100hu_$i.fst.windowed.weir.fst &amp;&amp; wc-l cw100hu_1.fst.windowed.weir.fst ;done<br>for i in {1..31}; do head -n 100000 cw100hu_$i.fst.windowed.weir.fst &gt; topw100hu_$i.fst.windowed.weir.fst &amp;&amp; cat topw100hu_$i.fst.windowed.weir.fst &gt; topchr1.txt ;done</p>
<p>##优先分群体</p>
<p>for i in {1..10}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 150000 –fst-window-step 75000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;windowsnp&#x2F;fst&#x2F;hu_$i ;done<br>for i in {11..20}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 150000 –fst-window-step 75000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;windowsnp&#x2F;fst&#x2F;hu_$i ;done<br>for i in {21..31}; do vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –fst-window-size 150000 –fst-window-step 75000 –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;windowsnp&#x2F;fst&#x2F;hu_$i ;done</p>
<p>for i in {1..31}; do cd &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window2&#x2F;100&#x2F;group$i &amp;&amp; cat *fst &gt; m.txt &amp;&amp; awk ‘{if($6<del>&#x2F;e&#x2F;)$6&#x3D;0}1’ m.txt &gt; mm.txt &amp;&amp; awk ‘{if($6</del>&#x2F;nan&#x2F;)$6&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k6 mmm.txt &gt; mmmm.txt &amp;&amp;  head -n 1500 mmmm.txt &gt; &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window2&#x2F;100&#x2F;top&#x2F;topgroup$i.txt ;done</p>
<p>for i in {1..15}; do cd &#x2F;home&#x2F;l&#x2F;data&#x2F;fstsnp&#x2F;fsthugroup$i &amp;&amp; cat *fst &gt; m.txt &amp;&amp; awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; sort.txt ;done<br>for i in {1..31}; do cd &#x2F;home&#x2F;l&#x2F;data&#x2F;fstsnp&#x2F;fsthugroup$i &amp;&amp; cat *fst &gt; m.txt &amp;&amp; awk ‘{if($3<del>&#x2F;e&#x2F;)$3&#x3D;0}1’ m.txt &gt; mm.txt &amp;&amp; awk ‘{if($3</del>&#x2F;nan&#x2F;)$3&#x3D;0}1’ mm.txt &gt; mmm.txt &amp;&amp; sort -nr -k3 mmm.txt &gt; sort.txt ;done</p>
<p>cat ptop* &gt; mptop.txt<br>wc -l mptop.txt<br>sort -nr -k6 mergetop.txt &gt; sortmergetop.txt<br>tail -n sortmergetop.txt<br>awk ‘{ print $1 “_” $2 “\t” $0 }’ laji.txt &gt; maptop.txt &amp;&amp; cut -f 1,2,3 maptop.txt &gt; macptop.txt &amp;&amp; sort macptop.txt  | uniq -c &gt;  macptopout.txt</p>
<p>awk ‘{ if ($1 &gt; 8) count++ } END { print count }’ mmmm.txt<br> head -n 12340 mmmm.txt &gt; &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window2&#x2F;100&#x2F;top&#x2F;topgroup1.txt</p>
<p>sh getwindow.sh &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window2&#x2F;100&#x2F;top&#x2F;vcf&#x2F;sort24.txt &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;fst_window2&#x2F;100&#x2F;top&#x2F;vcf&#x2F;hu.vcf</p>
<p>nohup  vcftools –vcf caid.vcf –remove &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –recode –recode-INFO-all  –out othergroups &amp;<br>nohup vcftools –vcf caid.vcf –keep &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –recode –recode-INFO-all  –out hu &amp;</p>
<p>  vcftools –vcf hu.recode.vcf –diff other_groups.recode.vcf –diff-site –out diff_stats</p>
<p>  awk ‘{ if ($5 &gt; 0.6999990) count++ } END { print count }’ mergedd.txt<br>  awk ‘{ if ($5 &gt; 0.7999999) count++ } END { print count }’ mergedd.txt<br>  awk ‘{ if ($5 &lt; -0.6999999) count++ } END { print count }’ mergedd.txt<br>  awk ‘{ if ($5 &lt; -0.7999999) count++ } END { print count }’ mergedd.txt</p>
<p>  cut -f 1,2,6 hu.freq.frq &gt; hufreq.txt</p>
<p>  awk -F ‘:’ ‘{print $2}’ hufreq.txt &gt; huf.txt</p>
<p>paste -d ‘\t’ hufreq.txt huf.txt &gt; merged.txt</p>
<p> for i in {1..31}; do cd &#x2F;home&#x2F;l&#x2F;data&#x2F;fstsnp&#x2F;fsthugroup$i &amp;&amp; cat *0.8.txt &gt; cayi$i.txt &amp;&amp; cp cayi$i.txt &#x2F;home&#x2F;l&#x2F;data&#x2F;fstsnp&#x2F;0.8snp ;done</p>
<p> for i in {1..31}; do cut -f 1,2 cayi$i.txt &gt; cayisite$i.txt ;done<br> cat cayisite* &gt; allcayisite.txt &amp;&amp; sort -k1n,1 -k2n,2 allcayisite.txt &gt; sortcayisite.txt &amp;&amp; uniq -c sortcayisite.txt &gt; result.txt </p>
<p> awk ‘{split($1, arr, “ “); print arr[1] “\t” arr[2] “\t” $2 “\t” }’ result.txt &gt; b.txt &amp;&amp; sort -nr -k 1 b.txt &gt; c.txt<br>  awk ‘{split($1, arr, “ “); print arr[1] “\t” arr[2] “\t” $2 “\t” $2 “\t” $3 }’ usc6b5.txt &gt; fusc6b5.txt<br>  cat b5.txt c6.txt &gt; c6a.txt &amp;&amp; sort -k1n,1 -k2n,2 c6a.txt &gt; sc6a.txt &amp;&amp; uniq -c sc6a.txt &gt; usc6a.txt &amp;&amp;   awk ‘{split($1, arr, “ “); print arr[1] “\t” arr[2] “\t” $2 “\t” $3  }’ usc6a.txt &gt; fusc6a.txt &amp;&amp; sort -nr -k 1 fusc6a.txt &gt; sfusc6a.txt</p>
<p>  bcftools view -T b.txt hu.vcf.gz -o selected_snps.vcf.gz</p>
<p>  for i in {1..31} ; do awk ‘{if($5<del>&#x2F;e&#x2F;)$5&#x3D;0}1’ hu_$i.windowed.weir.fst &gt; vhu_$i.windowed.weir.fst &amp;&amp; awk ‘{if($5</del>&#x2F;nan&#x2F;)$5&#x3D;0}1’ vhu_$i.windowed.weir.fst &gt; vvhu_$i.windowed.weir.fst &amp;&amp; sort -nr -k 5 vvhu_$i.windowed.weir.fst &gt; vvvhu_$i.windowed.weir.fst &amp;&amp; head -n 100000 vvvhu_$i.windowed.weir.fst &gt; tophu_$i.windowed.weir.fst ;done</p>
<p> for i in {1..3} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {4..8} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g2chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg2chr$i ;done<br> for i in {9..13} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g2chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg2chr$i ;done<br> for i in {14..17} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g2chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg2chr$i ;done<br> for i in {18..22} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g2chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg2chr$i ;done<br> for i in {23..27} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g2chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg2chr$i ;done</p>
<p> for i in {1..2} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {3..5} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {6..8} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {9..11} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br>for i in {12..13} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {13..15} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {16..18} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {19..21} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br>for i in {22..24} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done<br> for i in {25..27} ; do java -Xms100g -Xmx100g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100 gt&#x3D;g1chr$i.recode.vcf map&#x3D;chr_$i.map out&#x3D;phasingg1chr$i ;done</p>
<p>  awk ‘{split($1, arr, “ “); print arr[1] arr[2] “\t” $2 “\t” $3 “\t” $4 }’ jiaoji9.txt &gt; a9 &amp;&amp; cut -f 2,3,4 a9 &gt; b9 &amp;&amp; sort -nr -k1n,1 -k2n,2  b9 &gt; c9.txt &amp;&amp; uniq -c c9 &gt; d9 &amp;&amp; wc -l d15<br>cut -f 1 b9 &gt; c9 &amp;&amp; sort c9 &gt; e9 &amp;&amp; uniq -c e9 &gt; d9 &amp;&amp; wc -l d9<br>nohup java -Xms200g -Xmx200g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;120 gt&#x3D;g2id.vcf map&#x3D;new.map out&#x3D;phasingg2 &amp;</p>
<p>for i in {1..27}; do java -Xms400g -Xmx400g -jar  &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;120 gt&#x3D;chr$i.recode.vcf out&#x3D;otherchr$i ; done<br>for i in {1..27};do selscan –xpehh –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;cayi&#x2F;phasing&#x2F;chrhu&#x2F;huchr$i.vcf.gz –vcf-ref &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;cayi&#x2F;phasing&#x2F;chrother&#x2F;otherchr$i.vcf.gz  –map &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;cayi&#x2F;chrmap&#x2F;chr_$i.map –out &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;cayi&#x2F;phasing&#x2F;xpehh&#x2F;hu_otherchr$i –threads 127 ;done<br>for i in {1..27}; do norm –xpehh –files &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;cayi&#x2F;phasing&#x2F;xpehh&#x2F;hu_otherchr$i.xpehh.out  –bp-win –winsize 150000  ;done</p>
<p>vcftools –vcf lxguolv.vcf.recode.vcf –weir-fst-pop g1.txt –weir-fst-pop g2.txt  –fst-window-size 150000 –fst-window-step 75000 –out g1_g2 ;done<br>vcftools –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop pop1.txt –weir-fst-pop pop2.txt –positions window_positions.txt –out output</p>
<p>for i in {1..31}; do vcftools  –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group$i.txt  –positions p12.txt –out c9hu_$i ;done</p>
<p>&#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf</p>
<p>  awk ‘{split($1, arr, “_”); print arr[1] “\t” $2 “\t” $3 “\t” $4 “\t” $5 “\t” $6 }’ treemix.tfam &gt; 1.txt</p>
<p>awk ‘{split($1,arr,”_”);print arr[1] “\t”  $1 “\t” $3 “\t” $4 “\t” $5 “\t” $6 }’ sample.tfam &gt; 3.txt</p>
<p>for m in {1..5}<br>do<br>    {<br>    for i in {1..5}<br>    do<br>        {<br>           treemix -i sample.treemix.in.gz -o sample.${m}.${i} -bootstrap   -root MG -m ${m} -k 1000 -noss -se -global<br>        }<br>    done<br>    }<br>done</p>
<p>###过滤缺失数据<br>vcftools –vcf lxguolv.vcf.recode.vcf –max-missing 0.7 –minDP 4  –recode –out SNPs_no_missing.vcf</p>
<h3 id="根据LD来过滤连锁的SNP"><a href="#根据LD来过滤连锁的SNP" class="headerlink" title="根据LD来过滤连锁的SNP"></a>根据LD来过滤连锁的SNP</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plink --vcf flooding_no_missing.vcf --indep-pairwise 50 10 0.2 --out tmp.ld --allow-extra-chr --set-missing-var-ids @:# --keep-allele-order</span><br><span class="line">~/biosoft/plink --vcf flooding_no_missing.vcf --extract tmp.ld.prune.in --freq --missing --within new_pop.cluster --out input --allow-extra-chr --set-missing-var-ids @:# --keep-allele-order</span><br><span class="line"></span><br><span class="line">plink --vcf $File --indep-pairwise 100 20 0.2 -out $&#123;File%%.*&#125;_LD100_20_02 --make-bed</span><br><span class="line">plink --bfile $&#123;File%%.*&#125;_LD100_20_02 --extract $&#123;File%%.*&#125;_LD100_20_02.prune.in --out $&#123;File%%.*&#125;_LD100_20_02_vcf --recode vcf-iid</span><br></pre></td></tr></table></figure>

<p>for m in {1..5}<br>   do<br>   for i in {1..20}<br>      do<br>      treemix -se -bootstrap <br>         -i treemix_in.gz <br>         -o TreeMix.${i}.${m} <br>         -global <br>         -m ${m} <br>         -k 1000<br>      done<br>done</p>
<p>for m in {1..2}<br>do<br>    {<br>    for i in {1..5}<br>    do<br>        {<br>            nohup time treemix -i sample.treemix.in.gz -o sample.${m}.${i} -bootstrap   -root MG -m ${m} -k 1000 -noss -se -global 2&gt;&amp;1 &amp;<br>        }<br>    done<br>    }<br>done</p>
<p>for m in {3..4}<br>do<br>    {<br>    for i in {1..5}<br>    do<br>        {<br> treemix -i sample.treemix.in.gz -o sample.${m}.${i} -bootstrap 100 -root MG -m ${m} -k 500 -noss		<br> }<br>    done<br>    }<br>done</p>
<p>&#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;gatk&#x2F;gatk –java-options “-Xmx200g -XX:ParallelGCThreads&#x3D;50” CombineGVCFs -R &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;ref&#x2F;oarv4&#x2F;ref.fna -V tsp-b8-4-10.vcf -V tsp-b8-4-14.vcf -V  tsp-b8-4-11.vcf -V  tsp-b8-4-12.vcf  -O merged.vcf.gz</p>
<p>&#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;gatk&#x2F;gatk VariantFiltration <br>    -V merged.vcf.gz <br>    -filter “QD &lt; 2.0” –filter-name “QD2” <br>    -filter “QUAL &lt; 30.0” –filter-name “QUAL30” <br>    -filter “SOR &gt; 3.0” –filter-name “SOR3” <br>    -filter “FS &gt; 60.0” –filter-name “FS60” <br>    -filter “MQ &lt; 40.0” –filter-name “MQ40” <br>    -filter “MQRankSum &lt; -12.5” –filter-name “MQRankSum-12.5” <br>    -filter “ReadPosRankSum &lt; -8.0” –filter-name “ReadPosRankSum-8” <br>    -O snps_filtered.vcf.gz</p>
<p>java -Xms400g -Xmx400g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar combineGprobs -prefix merged  -nthreads&#x3D;80 -chr phasingg1chr1.vcf.gz phasingg1chr2.vcf.gz<br>java -Xms400g -Xmx400g -jar &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;beagle&#x2F;beagle.22Jul22.46e.jar nthreads&#x3D;100   gt&#x3D;&#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf323&#x2F;use323.vcf out&#x3D;phase323vcf</p>
<p>vcftools  –vcf &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;vcf&#x2F;caid.vcf –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;group19.txt –weir-fst-pop &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;data&#x2F;grouph&#x2F;305.txt –fst-window-size 150000 –fst-window-step 75000   –out hu_305</p>
<p>for i in {1..27};do &#x2F;mnt&#x2F;d&#x2F;lx&#x2F;soft&#x2F;selscan&#x2F;bin&#x2F;linux&#x2F;norm –xpehh –files  hu_1chr$i.xpehh.out –bp-win –winsize 150000;done</p>

    </div>

    
    
    

    <footer class="post-footer">

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2024/04/28/xpehh/" rel="prev" title="">
                  <i class="fa fa-angle-left"></i> 
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2024/04/28/fst_sh/" rel="next" title="">
                   <i class="fa fa-angle-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">

  <div class="copyright">
    &copy; 
    <span itemprop="copyrightYear">2024</span>
    <span class="with-love">
      <i class="fa fa-heart"></i>
    </span>
    <span class="author" itemprop="copyrightHolder">John Doe</span>
  </div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/muse/" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

    </div>
  </footer>

  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>
  <div class="sidebar-dimmer"></div>
  <div class="back-to-top" role="button" aria-label="Back to top">
    <i class="fa fa-arrow-up fa-lg"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha256-XL2inqUJaslATFnHdJOi9GfQ60on8Wx1C2H8DYiN1xY=" crossorigin="anonymous"></script>
<script src="/js/comments.js"></script><script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/schemes/muse.js"></script><script src="/js/sidebar.js"></script><script src="/js/next-boot.js"></script>

  






  





</body>
</html>
